code/controllers/subsystem/movement/movement_types.dm

```diff
@@ -869,3 +869,93 @@
 	var/atom/old_loc = moving.loc
 	holder.current_pipe = holder.current_pipe.transfer(holder)
 	return old_loc != moving?.loc ? MOVELOOP_SUCCESS : MOVELOOP_FAILURE
+
+
+/**
+ * Helper proc for the smooth_move datum
+ *
+ * Returns TRUE if the loop sucessfully started, or FALSE if it failed
+ *
+ * Arguments:
+ * moving - The atom we want to move
+ * angle - Angle at which we want to move
+ * delay - How many deci-seconds to wait between fires. Defaults to the lowest value, 0.1
+ * timeout - Time in deci-seconds until the moveloop self expires. Defaults to INFINITY
+ * subsystem - The movement subsystem to use. Defaults to SSmovement. Only one loop can exist for any one subsystem
+ * priority - Defines how different move loops override each other. Lower numbers beat higher numbers, equal defaults to what currently exists. Defaults to MOVEMENT_DEFAULT_PRIORITY
+ * flags - Set of bitflags that effect move loop behavior in some way. Check _DEFINES/movement.dm
+ *
+**/
+
+/datum/move_manager/proc/smooth_move(moving, angle, delay, timeout, subsystem, priority, flags, datum/extra_info)
+	return add_to_loop(moving, subsystem, /datum/move_loop/smooth_move, priority, flags, extra_info, delay, timeout, angle)
+
+/datum/move_loop/smooth_move
+	/// Angle at which we move. 0 is north because byond.
+	var/angle = 0
+	/// When this gets bigger than 1, we move a turf
+	var/x_ticker = 0
+	var/y_ticker = 0
+	/// The rate at which we move, between 0 and 1. Cached to cut down on trig
+	var/x_rate = 0
+	var/y_rate = 1
+	/// Sign for our movement
+	var/x_sign = 1
+	var/y_sign = 1
+	/// Actual speed, in case we move faster than 1 tile per tick
+	var/speed = 1
+
+/datum/move_loop/smooth_move/setup(delay, timeout, angle)
+	. = ..()
+	if(!.)
+		return FALSE
+	set_angle(angle)
+
+/datum/move_loop/smooth_move/compare_loops(datum/move_loop/loop_type, priority, flags, extra_info, delay, timeout, atom/chasing, home = FALSE)
+	if(..() && angle == src.angle)
+		return TRUE
+	return FALSE
+
+/datum/move_loop/smooth_move/proc/set_speed(new_speed)
+	speed = new_speed
+	delay = max(1, round(new_speed, 0.1))
+
+/datum/move_loop/smooth_move/move()
+	var/atom/old_loc = moving.loc
+	var/times_to_loop = 1
+	if (speed < 1)
+		times_to_loop = 1 / speed
+
+	for (var/i in 1 to times_to_loop)
```

Comment:
why are you moving more then one turf per tick

Reply:
Because spacedrift can accelerate you up to 2 turfs per tick, is that a bad idea?